### PR TITLE
Ready for 3.7 to update core

### PR DESCRIPTION
**Pull request by degobbis**
Bugfix to improve "undefined notice" in fof
Bugfix on updateCore(), when using a different temp_path in the configuration as joomla default
Bugfix on updateCore(), to prevent the installation of core files, if no update is available

### CODE REVIEW

**Comment by zero-24** [comment]
> Bugfix to improve "undefined notice" in fof

Another time another issue with fof :(

File: src/cli/update.php [hunk 1]
```diff
@@ -309,12 +316,13 @@ public function updateCore()
...
 		$updateInformation = $jUpdate->getUpdateInformation();
 
-		if (!empty($updateInformation['object']))
+		if (!empty($updateInformation['hasUpdate']))
 		{
 			$packagefile = JInstallerHelper::downloadPackage($updateInformation['object']->downloadurl->_data);
-			$packagefile = JPATH_BASE . '/tmp/' . basename($packagefile);
-			$package = JInstallerHelper::unpack($packagefile, true);
-			JFolder::copy($package['extractdir'], JPATH_BASE, '', true, true);
+			$tmp_path    = $this->app->getCfg('tmp_path');
```
Comment (zero-24) [inline src/cli/update.php]: $this->app->get('tmp_path');

Comment (degobbis) [reply src/cli/update.php]: I know, but it it will be removed in 4.0 and to the purpose of the downward compatibility...

Comment (zero-24) [reply src/cli/update.php]: > I know, but it it will be removed in 4.0 and to the purpose of the downward compatibility...

?

**Comment by degobbis** [comment]
@rdeutz thx

**Pull request merged**